### ISSUE TITLE
Impossible to have multiple instances

### ISSUE THREAD

**Issue by hrobertson** [miragejs/miragejs#414]
My app talks to multiple APIs.
I want to be able to mock part or all of any subset of those APIs.

After discovering this didn't work out of the box, I thought passing a single pretender instance to each instance of Server would solve it, but no.

If I create once instance for api-a and another for api-b, then requests to api-a get mocked fine. Requests to api-b result in:
> Mirage: Your app tried to GET 'http://api-b/', but there was no route defined to handle this request. Define a route for this endpoint in your routes() config. Did you forget to define a namespace?

If I add `this.passthrough('https://api-b/**')` to my api-a routes(), then calls to api-b get passed through and don't get intercepted by the api-b mock.

The only solution I can think of is to write some logic to merge all these into one miragejs Server instance but that's obviously undesirable.

I can't find any mention of this limitation in the docs. Am I missing something that can get this working? If not, is this something that might be worked on in the future?

Thanks

**Comment by samselikoff** [comment]
Yeah this is a limitation due to the way Pretender works (global monkey patching), but Mirage server instances have a `config()` option you can use to "augment" an existing server. You should be able to call that from different files to add new routes.

```
// app.js
export const server = new Server

// my-component.js
import { server } from './app'

server.config({
  routes() {
    this.get('/foo', () => {})
  }
})
```

Could you let me know if that works? I think it should.

We definitely need to add this to the docs!

**Comment by hrobertson** [comment]
Thanks for the suggestion. I'll revisit this at some point as I've shelved it for now and it would involve some refactoring (my mocks are wrapped in other classes which handle config and state).

Also I'm using typescript and just updated to miragejs 0.1.37 which includes types... and the Server type doesn't include the config method. Presumably because it's kind of a private method.

**Comment by Apetree100122** [comment]
````JS
// app.export 
const server = new Server
// my-component.js
import { server
 } from './app'
server.config(
{ routes(a/b) 
{  this.get('
/foo', (     
) => {}) }
})
````
